tools/remove_grouping_from_report_results.rb

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+require 'trollop'
+
+ARGV.shift if ARGV[0] == '--' # if invoked with rails runner
+opts = Trollop.options do
+  banner "Remove extras[:grouping] from MiqReportResult#report column.\n\nUsage: ruby #{$0} [options]\n\nOptions:\n\t"
+  opt :dry_run,    "For testing, rollback any changes when the script exits.", :short => :none, :default => false
+  opt :batch_size, "Limit memory usage by process this number of report results at a time.",    :default => 50
+  opt :count,      "Stop checking after this number of report results.",                        :default => 0
+end
+
+puts "Using options: #{opts.inspect}"
+
+# Load rails after trollop options are set.  No one wants to wait for -h.
+require File.expand_path('../config/environment', __dir__)
+if opts[:dry_run]
+  puts "Running in dry-run, changes will be rolled back when complete."
+  ActiveRecord::Base.connection.begin_transaction(joinable: false)
+
+  at_exit do
+    ActiveRecord::Base.connection.rollback_transaction
+  end
+end
+
+start = Time.now
+total = 0
+fixed = 0
+MiqReportResult.find_each(batch_size: opts[:batch_size]).with_index do |rr, i|
+  break if opts[:count] > 0 && i == opts[:count]
+  report = rr.report
+  next if report.nil? || report.extras.nil?
+
+  if report.extras.key?(:grouping)
+    rr.report.extras = report.extras.except!(:grouping)
```

Comment:
Pendantic:  If you wanted to use less code, you could just do a `rr.report.extras.delete(:grouping)`.

Reply:
Truth.  I'll fix.